packages/blade/src/components/ButtonGroup/StyledButtonGroup.tsx

```diff
@@ -2,33 +2,38 @@ import styled from 'styled-components';
 import type { StyledButtonGroupProps } from './types';
 import BaseBox from '~components/Box/BaseBox';
 import { makeBorderSize } from '~utils';
-import { getBackgroundColorToken } from '~components/Button/BaseButton/BaseButton';
-import getIn from '~utils/lodashButBetter/get';
 
 const StyledButtonGroup = styled(BaseBox)<StyledButtonGroupProps>(
-  ({ theme, isDisabled, variant = 'primary', color, isFullWidth }) => {
+  ({ theme, variant = 'primary', isFullWidth }) => {
     return {
       display: 'flex',
       width: isFullWidth ? '100%' : 'fit-content',
       borderWidth: makeBorderSize(theme.border.width.thin),
       borderRadius: makeBorderSize(theme.border.radius.medium),
       borderStyle: 'solid',
-      borderColor: getIn(
-        theme.colors,
-        getBackgroundColorToken({
-          // Only secondary variant has border a border, for other variants we use background color so that the border is not visible
-          property: variant === 'secondary' ? 'border' : 'background',
-          variant,
-          color,
-          state: isDisabled ? 'disabled' : 'default',
-        }),
-      ),
-
+      borderColor: 'transparent',
+      overflow: 'hidden',
       'button[role="button"]': {
-        borderRadius: 0,
-        border: 'none',
         flex: isFullWidth ? 1 : 'auto',
+        borderRadius: 0,
       },
+
+      ...(variant === 'secondary' && {
+        'button[role="button"]:first-child': {
+          borderRight: 'none',
+          borderTopLeftRadius: makeBorderSize(theme.border.radius.medium),
+          borderBottomLeftRadius: makeBorderSize(theme.border.radius.medium),
+        },
+        'button[role="button"]:not(:first-child):not(:last-child)': {
+          borderLeft: 'none',
+          borderRight: 'none',
+        },
+        'button[role="button"]:last-child': {
+          borderLeft: 'none',
+          borderTopRightRadius: makeBorderSize(theme.border.radius.medium),
+          borderBottomRightRadius: makeBorderSize(theme.border.radius.medium),
+        },
```

Comment:
How is this related to loading? 🤔

Reply:
Earlier, we added a border to the Button Group container and removed the borders from all individual buttons.
However, when we enabled the loading state for a button, it looked like this:

<img width="589" alt="Screenshot 2025-02-14 at 3 58 11 PM" src="https://github.com/user-attachments/assets/71d85df4-8562-4c0e-8e8a-9cdd55af7124" />
I think the ideal way to handle this was to modify the border-radius of the buttons. So, I refactored the Button Group code a bit and also updated how the border is handled for the secondary variant.